### PR TITLE
Enabling sleef to be referenced with gnu vector abi

### PR DESCRIPTION
**Pull request by shibatch**
Please check the current state.

### CODE REVIEW

File: src/libm/sleefsimddp.c [hunk 1]
```diff
@@ -1751,6 +1784,7 @@ EXPORT CONST vdouble2 xmodf(vdouble x) {
 
   return ret;
 }
+#endif
```
Comment (fpetrogalli-arm) [inline src/libm/sleefsimddp.c]: `//endif ENABLE_GNUABI`

File: src/libm/sleefsimdsp.c [hunk 2]
```diff
@@ -1449,6 +1478,7 @@ EXPORT CONST vfloat2 xmodff(vfloat x) {
 
   return ret;
 }
+#endif
```
Comment (fpetrogalli-arm) [inline src/libm/sleefsimdsp.c]: `//#ifndef ENABLE_GNUABI`

File: src/script/abspath.sh [hunk 3]
```diff
@@ -0,0 +1,9 @@
+#!/bin/sh
+#!/bin/sh
+if [ `uname` = Darwin ]; then
+    find `cd $1; pwd` -name $2 -print
```
Comment (fpetrogalli-arm) [inline src/script/abspath.sh]: This is exactly the same as the one in line 8. Please merge it into the `#else` statement.

File: src/libm/Makefile [hunk 4]
```diff
@@ -26,34 +26,44 @@ endif
...
 
 libm.a : $(OBJ)
 	$(FLOCK) libm.a.lock -c 'ar rcs libm.a $(OBJ)'
 
+libgm.a : $(OBJ2)
+	$(FLOCK) libgm.a.lock -c 'ar rcs libgm.a $(OBJ2)'
+
+OBJ.txt : $(OBJ)
+	rm -rf OBJ.txt
+	for i in `echo $(OBJ)`; do ../script/abspath.sh $(@D) $$i >> OBJ.txt; done
```
Comment (fpetrogalli-arm) [inline src/libm/Makefile]: I think you can replace this line with the following
```
for i in `echo $(OBJ)`; do realpath $$i >> OBJ.txt; done
```

File: src/libm/Makefile [hunk 5]
```diff
@@ -26,34 +26,44 @@ endif
...
-all : libm.a sleef.h
+all : libm.a libgm.a sleef.h
 
 libm.a : $(OBJ)
 	$(FLOCK) libm.a.lock -c 'ar rcs libm.a $(OBJ)'
 
+libgm.a : $(OBJ2)
+	$(FLOCK) libgm.a.lock -c 'ar rcs libgm.a $(OBJ2)'
+
+OBJ.txt : $(OBJ)
```
Comment (fpetrogalli-arm) [inline src/libm/Makefile]: Why are you generating this text file? Same question for OBJ2.txt

File: src/libm-tester/iutsimdmain.c [hunk 6]
```diff
@@ -0,0 +1,57 @@
+//          Copyright Naoki Shibata 2010 - 2017.
```
Comment (fpetrogalli-arm) [inline src/libm-tester/iutsimdmain.c]: Nice. Can you remove the old `tester.c` now?

File: src/libm/sleefsimdsp.c [hunk 7]
```diff
@@ -412,6 +382,63 @@ EXPORT CONST vfloat xcosf_u1(vfloat d) {
   return u;
 }
 
+#ifndef ENABLE_GNUABI
```
Comment (fpetrogalli-arm) [inline src/libm/sleefsimdsp.c]: Why not change the prototype when enabling GNUABI instead of removing the function?

File: src/libm/Makefile (hunk 5 above)
Comment (shibatch) [inline src/libm/Makefile]: Please see Makefile under lib directory.
In order to build the DLLs for OSX and Windows, I need a list of absolute paths of the object files.
Please tell me if there is a better way.

File: src/libm/Makefile (hunk 4 above)
Comment (shibatch) [inline src/libm/Makefile]: realpath is not available on Mac OS X.

File: src/libm-tester/iutsimdmain.c (hunk 6 above)
Comment (shibatch) [inline src/libm-tester/iutsimdmain.c]: tester.c is still in use. Do you mean another file?
I think I need all files.

File: src/script/abspath.sh (hunk 3 above)
Comment (shibatch) [inline src/script/abspath.sh]: I wrote in this way in order to make it work on Mac.
On Mac, 'uname -o' does not work. != operator in [ does not work either.

Comment (fpetrogalli-arm) [inline src/script/abspath.sh]: Make sense, thanks for the explanation.

File: src/libm-tester/iutsimdmain.c (hunk 6 above)
Comment (fpetrogalli-arm) [inline src/libm-tester/iutsimdmain.c]: I thought this was a replacement of `tester.c`. If not, why did you add it?

File: src/libm/Makefile (hunk 5 above)
Comment (fpetrogalli-arm) [inline src/libm/Makefile]: I see, thanks for the explanation. 

I am not aware of  a better way of doing this other than using`cmake` instead of using hand crafted Makefiles. Keep it as it is, we will think about moving to cmake in the future.

File: src/script/abspath.sh (hunk 3 above)
Comment (fpetrogalli-arm) [inline src/script/abspath.sh]: Thanks for the explanation.

File: src/libm-tester/iutsimdmain.c (hunk 6 above)
Comment (shibatch) [inline src/libm-tester/iutsimdmain.c]: Below is what I wrote in the commit log,

   iutsimd.c is now divided into two parts. This is needed to make it
   work with icc. In the previous implementation, the whole program is
   compiled with support for extended ISA (e.g. AVX512F), and this
   caused the IUT not to exit with the correct exit code when that
   ISA is not supported on the host.

Comment (fpetrogalli-arm) [inline src/libm-tester/iutsimdmain.c]: I see. Thanks for the explanation. It is easy to miss important bits of information when the patches are big and cover multiple aspects. It would be better if you could submit pull request for small incremental changes.

Comment (shibatch) [inline src/libm-tester/iutsimdmain.c]: I am sorry for that, but it would be too much overhead. I will have smaller amount of time I can spend for this project after June, so I think that would solve this problem.

**Comment by fpetrogalli-arm** [comment]
LGTM. Please merge.
Thank you!
